osquery/tables/system/windows/prefetch.cpp

```diff
@@ -0,0 +1,377 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include "osquery/filesystem/fileops.h"
+#include <osquery/core/core.h>
+#include <osquery/core/tables.h>
+#include <osquery/filesystem/filesystem.h>
+#include <osquery/logger/logger.h>
+#include <osquery/tables/system/windows/prefetch.h>
+#include <osquery/utils/conversions/join.h>
+#include <osquery/utils/conversions/tryto.h>
+#include <osquery/utils/conversions/windows/strings.h>
+#include <osquery/utils/conversions/windows/windows_time.h>
+#include <osquery/utils/windows/lzxpress.h>
+
+#include <boost/algorithm/hex.hpp>
+#include <boost/filesystem.hpp>
+
+namespace osquery {
+namespace tables {
+
+const std::string kPrefetchLocation = getSystemRoot().string() + "\\Prefetch\\";
```

Comment:
Preference would be to use PathAppend or something equivalent here to join these two strings. Not sure SystemRoot will ever end with a backslash though, so what you have might be okay.

Reply:
`getSystemRoot` returns C:\Windows, with no backslash.  
I checked other tables using `getSystemRoot` and they also append `+ "\\"` to the path.  
So it looks like `getSystemRoot` shouldnt return a backslash?

Reply:
My concern here is that we're assuming that `getSystemRoot` will always return with no backslash, and if it was safe to make that assumption. :)

I fear it is not a safe assumption, since `getSystemRoot` calls `GetWindowsDirectory`, and this function can in fact return with a backslash if installed in the root directory. Therefore, PathAppend, or some equivalent path-joining function, is a better here.

Reply:
In fact, I see that `getSystemRoot` has an issue since it's assuming that `GetWindowsDirectory` will always succeed....

Reply:
this should be fixed

Reply:
Thank you! :) Sorry, I'm juggling many different tasks, which is why my response is a bit slower.